cilium-health/launch/endpoint.go

```diff
@@ -262,11 +262,23 @@ func LaunchAsEndpoint(owner endpoint.Owner, hostAddressing *models.NodeAddressin
 		return fmt.Errorf("Error while configuring routes: %s", err)
 	}
 
-	// Add the endpoint
 	if err := endpointmanager.AddEndpoint(owner, ep, "Create cilium-health endpoint"); err != nil {
 		return fmt.Errorf("Error while adding endpoint: %s", err)
 	}
 
+	if !ep.SetStateLocked(endpoint.StateWaitingToRegenerate, "initial build of health endpoint") {
```

Comment:
The endpoint has been added into the endpointmanager at this time, and as such is exposed to the rest of the daemon for manipulation / changes. The endpoint lock should be acquired here.